src/iris/api.py

```diff
@@ -3900,6 +3922,52 @@ def on_put(self, req, resp, username):
         resp.status = HTTP_204
 
 
+class UserTemplateOverrides(object):
+    allow_read_no_auth = False
+    enforce_user = True
+
+    def on_post(self, req, resp, username):
+        # check request body integrity
+        try:
+            data = ujson.loads(req.context['body'])
+        except ValueError:
+            raise HTTPBadRequest('Invalid json in post body')
+        if not isinstance(data.get('template_overrides'), dict):
+            raise HTTPBadRequest('Invalid json in post body')
```

Comment:
Use a unique error string for each error so that it's easily identifiable where in the code the exception occurred. 

I think this one can be something like `"Invalid json in post body: 'template_overrides' is not a dict"` or something like that

Reply:
sure, changed